kmm/kmm-feed/src/commonMain/kotlin/com/gchristov/newsfeed/kmmfeed/FeedViewModel.kt

```diff
@@ -20,8 +20,34 @@ class FeedViewModel(
     dispatcher = dispatcher,
     initialState = State()
 ) {
+    private val searchQueryFlow = MutableStateFlow("")
+
     init {
         loadNextPage()
+        observeSearchQuery()
+    }
+
+    /**
+     * Observer function placed on the search text being updated from the UI.
+     *
+     * The MutableStateFlow, representing the sequence of keystrokes being emitted
+     * by the user, is debounced every 500ms so that downstream services (API,
+     * repository...) are not hit an unnecessary number of times and only when relevant
+     * results can be returned
+     */
+    private fun observeSearchQuery() {
+        launchUiCoroutine {
+            searchQueryFlow
+                .debounce(500)
+                .filter { text -> text.isNotEmpty() }
```

Comment:
this is here to avoid sending empty strings downstream - there's more to put here probably, but it can be added when API call is added?

Reply:
Good idea! Looks great